libfreerdp/core/nla.c

```diff
@@ -1085,6 +1085,53 @@ BOOL nla_decrypt_public_key_hash(rdpNla* nla)
 	return status;
 }
 
+static BOOL set_creds_octetstring_to_settings(WinPrAsn1Decoder* dec, WinPrAsn1_tagId tagId,
+                                              BOOL optional, size_t settingId,
+                                              rdpSettings* settings)
+{
+	if (optional)
+	{
+		WinPrAsn1Decoder ctxtDec;
+		WinPrAsn1_tagId itemTag;
+		if (!WinPrAsn1DecPeekContextualTag(dec, &itemTag, &ctxtDec) || (itemTag != tagId))
+			return TRUE;
+	}
+
+	BOOL error = FALSE;
+	WinPrAsn1_OctetString value;
```

Comment:
initialize with `={0};`

Reply:
not needed, the next call sets the value or fails